paparazzi/src/main/java/app/cash/paparazzi/internal/ImageUtils.kt

```diff
@@ -68,7 +65,7 @@ internal object ImageUtils {
       var message = "Unable to load golden thumbnail: $relativePath\n"
       message = saveImageAndAppendMessage(thumbnail, message, relativePath, failureDir)
       if (FAIL_ON_MISSING_THUMBNAIL) {
-        fail(message)
+        throw AssertionError(message)
```

Comment:
Have you considered using `AssertionFailedError` from https://github.com/ota4j-team/opentest4j ?

The reasoning would be that this is a testing framework, and is running on JVM so it fits the umbrella open test is trying to standardize.

Reply:
Seems like there should be a failure policy interface that callers of the SDK can supply with a default instance implementation that throws AE. I would not take a hard dependency on that project.

Reply:
Is this another ISE, btw? as the test is broken, not failed.

Reply:
I don't know. I thought this case was when you add a snapshot call and run verify without having done a record.

Reply:
Correct, per the other reasonings, I'd say this should also be an ISE.